clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClient.java

```diff
@@ -5,237 +5,114 @@
 
 package com.datastrato.gravitino.client;
 
-import com.datastrato.gravitino.MetalakeChange;
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.CatalogChange;
 import com.datastrato.gravitino.NameIdentifier;
-import com.datastrato.gravitino.SupportsMetalakes;
-import com.datastrato.gravitino.dto.requests.MetalakeCreateRequest;
-import com.datastrato.gravitino.dto.requests.MetalakeUpdateRequest;
-import com.datastrato.gravitino.dto.requests.MetalakeUpdatesRequest;
-import com.datastrato.gravitino.dto.responses.DropResponse;
-import com.datastrato.gravitino.dto.responses.MetalakeListResponse;
-import com.datastrato.gravitino.dto.responses.MetalakeResponse;
-import com.datastrato.gravitino.dto.responses.VersionResponse;
-import com.datastrato.gravitino.exceptions.MetalakeAlreadyExistsException;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.SupportsCatalogs;
+import com.datastrato.gravitino.exceptions.CatalogAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.NoSuchCatalogException;
 import com.datastrato.gravitino.exceptions.NoSuchMetalakeException;
 import com.google.common.base.Preconditions;
-import java.io.Closeable;
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Gravitino Client for interacting with the Gravitino API, allowing the client to list, load,
- * create, and alter Metalakes.
+ * Gravitino Client for an user to interacte with the Gravitino API, allowing the client to list,
+ * load, create, and alter Catalog.
  *
  * <p>It uses an underlying {@link RESTClient} to send HTTP requests and receive responses from the
  * API.
  */
-public class GravitinoClient implements SupportsMetalakes, Closeable {
+public class GravitinoClient extends GravitinoClientBase implements SupportsCatalogs {
 
   private static final Logger LOG = LoggerFactory.getLogger(GravitinoClient.class);
 
-  private final RESTClient restClient;
+  private final String metalake_name;
 
-  private static final String API_METALAKES_LIST_PATH = "api/metalakes";
-  private static final String API_METALAKES_IDENTIFIER_PATH = "api/metalakes/";
+  private GravitinoMetaLake metaLake = null;
```

Comment:
To ensure visibility for the double-check mechanism, we may need to use the volatile keyword.